go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -1660,28 +1741,66 @@ func (oc *DefaultNetworkController) addHairpinAllowACL() error {
 		match = fmt.Sprintf("(%s || %s)", v4Match, v6Match)
 	}
 
-	ingressACLIDs := oc.getNetpolDefaultACLDbIDs(string(knet.PolicyTypeIngress))
-	ingressACL := BuildACL("", types.DefaultAllowPriority, match,
+	ingressACLIDs := bnc.getNetpolDefaultACLDbIDs(string(knet.PolicyTypeIngress))
+	ingressACL := bnc.BuildACL("", types.DefaultAllowPriority, match,
 		nbdb.ACLActionAllowRelated, nil, lportIngress, ingressACLIDs.GetExternalIDs())
 
-	egressACLIDs := oc.getNetpolDefaultACLDbIDs(string(knet.PolicyTypeEgress))
-	egressACL := BuildACL("", types.DefaultAllowPriority, match,
+	egressACLIDs := bnc.getNetpolDefaultACLDbIDs(string(knet.PolicyTypeEgress))
+	egressACL := bnc.BuildACL("", types.DefaultAllowPriority, match,
 		nbdb.ACLActionAllowRelated, nil, lportEgressAfterLB, egressACLIDs.GetExternalIDs())
 
-	ops, err := libovsdbops.CreateOrUpdateACLsOps(oc.nbClient, nil, ingressACL, egressACL)
+	ops, err := libovsdbops.CreateOrUpdateACLsOps(bnc.nbClient, nil, ingressACL, egressACL)
 	if err != nil {
 		return fmt.Errorf("failed to create or update hairpin allow ACL %v", err)
 	}
 
-	ops, err = libovsdbops.AddACLsToPortGroupOps(oc.nbClient, ops, types.ClusterPortGroupName, ingressACL, egressACL)
+	ops, err = libovsdbops.AddACLsToPortGroupOps(bnc.nbClient, ops, types.ClusterPortGroupName, ingressACL, egressACL)
 	if err != nil {
 		return fmt.Errorf("failed to add ACL hairpin allow acl to port group: %v", err)
 	}
 
-	_, err = libovsdbops.TransactAndCheck(oc.nbClient, ops)
+	_, err = libovsdbops.TransactAndCheck(bnc.nbClient, ops)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
+
+// WatchNetworkPolicy starts the watching of multinetworkpolicy resource and calls
+// back the appropriate handler logic
+func (bnc *BaseNetworkController) WatchNetworkPolicy() error {
```

Comment:
why? they are doing the exactly the same thing.

Reply:
My general logic here is if
1. the function is called explicitly from default and secondary controllers, and not from `BaseNetworkController` methods
2. and it has different implementations for default and secondary controllers
then it should be 2 different functions.

in this case, `WatchNetworkPolicy` is called from the `Run` method of every controller explicitly, and has this `bnc.IsSecondary()` branching, so it may be 2 separate function for default and secondary controller

Also, `policyHandler` field is only required for secondary controllers, so it may also be moved from the `BaseNetworkController`. Then `WatchNetworkPolicy` for default network controller will stay exactly as before
```
func (oc *DefaultNetworkController) WatchNetworkPolicy() error {
	_, err := oc.retryNetworkPolicies.WatchResource()
	return err
}
```
and secondary controller `WatchNetworkPolicy` method may add all these extra checks and fields

Reply:
ok